# The Wilcoxon Tests
Let $X$ denote the number of seconds between two calls arrived in a specific call center and $m$ denote the median of $X$. A sample of 20 call time has been collected as follows:

(6.8,5.7,2.1,19.0,4.8,18.9,6.9,5.3,4.1,9.8,1.7,7.0,18.9,16.9,10.4,44.1,2.9,2.4,4.8,7.9)

Test the hypothesis $H_{0}: m = 6.2$ against the alternative $H_{1}: m < 6.2$.

In [38]:
n = 20
s = c(6.8,5.7,2.1,19.0,4.8,18.9,6.9,5.3,4.1,9.8,1.7,7.0,18.9,16.9,10.4,44.1,2.9,2.4,4.8,7.9)
s

[1]  6.8  5.7  2.1 19.0  4.8 18.9  6.9  5.3  4.1  9.8  1.7  7.0 18.9 16.9 10.4
[16] 44.1  2.9  2.4  4.8  7.9

## The Test Statistic
Define the random variable $Y_{i} = 1_{X_{i} - m_{0} < 0}(X_{i} - m_{0})$ and the test statistic $Y = \sum_{i=1}^{n}Y_{i}$. I follows that $Y \sim Binom(n, .5)$. The value of the test statistic of the sample is:

In [101]:
y = sum(s < 6.2)
y

[1] 9

## Density Plot of $Binom(n, .5)$

In [ ]:
x=seq(0, 20)
ps = dbinom(x, n, 1/2)
plot(ps, pch=19, xlab='n', ylab='probability', col='blue')

## The *p*-Value

In [65]:
pVal = pbinom(y - 1, n, 1/2, lower.tail=F)
round(pVal, digits=4)

[1] 0.7483

Since *p*-Value is high as .7483 we do not reject the null hypothesis.

## The Critical Value

In [99]:
alphas = pbinom(seq(0, 20), 20, 1/2, lower.tail=T)
round(alphas, digits=6)
critVal = which.min(abs(alphas - .05))
paste('critVal:', critVal)
paste('alphas:', round(alphas[minIndx], digits=4))

[1] 0.000001 0.000020 0.000201 0.001288 0.005909 0.020695 0.057659 0.131588
 [9] 0.251722 0.411901 0.588099 0.748278 0.868412 0.942341 0.979305 0.994091
[17] 0.998712 0.999799 0.999980 0.999999 1.000000

[1] "critVal: 7"

[1] "alphas: 0.8684"

In [84]:
pbinom(13, n, 1/2, lower.tail=F)

[1] 0.05765915

In [94]:
seq(0, 19)

[1]  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19